django/thunderstore/security/views.py

```diff
@@ -0,0 +1,36 @@
+import binascii
+import json
+
+from cryptography.hazmat.primitives._serialization import Encoding, PublicFormat
+from cryptography.hazmat.primitives.asymmetric.ec import ECDSA
+from cryptography.hazmat.primitives.hashes import SHA256
+from django.http import HttpResponse
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+from sentry_sdk import capture_exception
+
+from thunderstore.security.utils import fetch_and_setup_public_key
+
+
+@csrf_exempt
+@require_POST
+def secret_scanning_endpoint(request):
```

Comment:
Re-implement this as a class based views, and use Django REST Framework's built-in features for it. Make sure to exclude it from the API documentation however (there should be a way to do that, just check the docs of the library used for autogenerated API docs).